Sources/StreamChat/StateLayer/ChatState.swift

```diff
@@ -47,6 +49,17 @@ public final class ChatState: ObservableObject {
     /// Use load messages in ``Chat`` for loading more messages.
     @Published public private(set) var messages = StreamCollection<ChatMessage>([])
     
+    /// Access a message available locally by its id.
+    ///
+    /// - Note: This method does a local lookup of the message.
+    ///
+    /// - Parameter messageId: The id of the message available locally.
+    ///
+    /// - Returns: An instance of the locally available chat message
+    public func message(for messageId: MessageId) -> ChatMessage? {
+        dataStore.message(id: messageId)
+    }
```

Comment:
Can we just get the message in the local array for now? 🤔 Why do we need this again, I did not fully understood?

Most of our customers are not using this `DataStoreProvider`, and ideally it would be nice to deprecate it. 

If we do want to expose a way for customers to query data in the local db only, it would be nice to createa new entity for that, maybe `MessagesStore` or something like that. Exposing local db only methods in the State Layer can be confusing IMO 🤔

Reply:
This is used in the logic around jumping to a message in SwiftUI, maybe also in UIKit. In general, would be nice to have a way to access local message.
I don't think we need a new entity for that. But I agree with this "Exposing local db-only methods in the State Layer can be confusing IMO". My suggestion is to update the method to sth like `localMessage` or `cachedMessage`.

Reply:
The general summary of the DataStore is:
UIKit
- [message] jump to message 
- [current user] creating message actions
- [channel] message search

SwiftUI
- [message] jump to message

We use it pretty rarely which is nice. The less we expose it the better for future.

Sounds like the options are:
a) Rename it to `localMessage` for emphasising that it only looks up locally available messages (It is still slightly confusing since it currently looks up any message, not messages part of the channel. If we later move to non FRC driven message pagination and use API responses directly within ChatMessage, then this method would change to be channel specific internally. All in all, I think it is fine. I could add a console warning if we want to limit the current function to channel specific lookup for future sake).

`
let message = dataStore.message(id: messageId)
if message.cid != cid {
  log.error("Trying to access message not part of the channel")
  return nil
}
return message
`
Just to guard against misuse.

b) UI components take the hit and do the lookup using the `.messages` which internally will trigger MessageDTO>ChatMessage conversions since it is a LazyCachedMapCollection. That is not so great.

Reply:
yes, b) is the best option